pkg/eraser/eraser.go

```
@@ -151,6 +162,45 @@ func removeImages(c Client, targetImages []string) error {
 	return nil
 }
 
+func isExcluded(img string, idToTagListMap map[string][]string) bool {
+	// check if img excluded by digest
+	if _, contains := excluded[img]; contains {
+		return true
+	}
+
+	// check if img excluded by name
+	if len(idToTagListMap[img]) > 0 {
+		if _, contains := excluded[idToTagListMap[img][0]]; contains {
+			return true
+		}
+	}
+
+	r := regexp.MustCompile(`[a-z0-9]+([._-][a-z0-9]+)*\/\*\z`)
+
+	// look for excluded repository values
+	for key := range excluded {
+		// if excluded key ends in /*, check image with pattern match
+		if match := r.MatchString(key); match {
+			// store repository name
+			repo := strings.Split(key, "*")
+
+			// check if img is part of repo
+			if match2, _ := regexp.MatchString("^"+repo[0], img); match2 {
```

This function swallows errors. It should return `(bool, error)` and return an error here if there's a problem.

Additionally, using a repo name as a regex could be problematic, because dots are valid in repo names, but they have a special meaning (a single wildcard character) in a regex. If you want to do this, you'll have to replace any `.` character with `\.`

---

good point!